lib/web/plug/params_validator.ex

```diff
@@ -0,0 +1,142 @@
+# Copyright(c) 2015-2024 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule Antikythera.Plug.ParamsValidator do
+  @moduledoc """
+  Plug to validate request parameters.
+
+  ## Usage
+
+  First, generate parameter validation models using `Antikythera.ParamStringStruct` or `Antikythera.BodyJsonStruct`.
+
+      defmodule MyQueryParams do
+        use Antikythera.ParamStringStruct,
+          fields: [
+            item_id: Croma.PosInteger
+          ]
+      end
+
+  Then, you can install this plug to validate parameters, as shown below.
+
+      plug Antikythera.Plug.ParamsValidator, :validate, query_params: MyQueryParams
+
+  In the controller function, you can access the validated parameters as follows.
```

Comment:
I would be better to expressly describe that each parameter types assign validated variables to the same name under `conn.assigns.validated`

Reply:
Fixed at bf5ee47.

Reply:
Thank you for the suggestion. Fixed at f4a5ac1.